windowing_lib_test.py

```diff
@@ -0,0 +1,71 @@
+import windowing as wl
+import numpy as np
+import pytest
+
+
+sample_matrix = {'x': [1, 1, 1, 1, 1, 20, 20, 1, 1, 1], 'y': [1, 1, 1, 1, 1, 20, 20, 1, 1, 1], 'z': [
+    1, 1, 1, 1, 1, 20, 20, 1, 1, 1], 'nanos': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}
+windowed_data = [[[[1, 1, 1, 1, 1], [1, 1, 1, 1, 1, ], [1, 1, 1, 1, 1], [1, 2, 3, 4, 5]], [
+        [20, 20, 1, 1, 1], [20, 20, 1, 1, 1], [20, 20, 1, 1, 1], [6, 7, 8, 9, 10]]]]
+np_windowed_data = np.array(windowed_data, 'd')
+
+def test_create_windows():
+    
+    window_size = 5
+    stride = 5
+
+    test_windowed_data = wl.create_windows_from_dictmatrix(
+        sample_matrix, 4, window_size, stride)
+    assert (test_windowed_data == np_windowed_data).all()
+
+
+lift_windows = [[[20, 20, 1, 1, 1], [20, 20, 1, 1, 1],
+                 [20, 20, 1, 1, 1], [6, 7, 8, 9, 10]]]
```

Comment:
I remember seeing a comment from Gautam on this earlier, but I can't find it anymore. What's the reason for not making the input data a test-function local variable?

def test_find_initial_lift_times():
    lift_windows = [[[...]]]
    exp_start = 6
    exp....

Reply:
Well there are 2 tests using lift_windows specifically but for the others I did forget to move them into their specific function, but I will do that right now